### PR TITLE
[firebase_ml_vision] adds facial contours

### PR DESCRIPTION
**Pull request by p30arena**
## Description

modified .gitignore to ommit some vscode related directories.
added contours to both android and iOS.
I'm not sure about sanity of the iOS part.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database](https://github.com/flutter/flutter/issues). Indicate, which of these issues are resolved or fixed by this PR. Note that you'll have to prefix the issue numbers with flutter/flutter#.*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by p30arena** [comment]
@bparrishMines 

These files are not formatted correctly (see diff above):
  firebase_ml_vision/android/src/main/java/io/flutter/plugins/firebasemlvision/FaceDetector.java
  firebase_ml_vision/ios/Classes/FaceDetector.m

```
> pub global run flutter_plugin_tools format
Downloading Google Java Format...
Formatting all .dart files...
CreateProcessW failed 2
Unhandled exception:
ProcessException: The system cannot find the file specified.
```

**Comment by p30arena** [comment]
I had to run the formatters manually

**Comment by p30arena** [comment]
contours should never be null
but I'm getting this error (maybe on iOS side?)

```
00:07 +18 -2: /tmp/cirrus-ci-build/packages/firebase_ml_vision/test/firebase_ml_vision_test.dart: FirebaseVision FaceDetector processImage with null landmark [E]                                      
  NoSuchMethodError: The method '[]' was called on null.
  Receiver: null
  Tried calling: []("allPoints")
  dart:core                                                 Object.noSuchMethod
  package:firebase_ml_vision/src/face_detector.dart 167:53  new Face._.<fn>
  dart:collection                                           new LinkedHashMap.fromIterables
  package:firebase_ml_vision/src/face_detector.dart 164:19  new Face._
  package:firebase_ml_vision/src/face_detector.dart 85:22   FaceDetector.processImage
  ===== asynchronous gap ===========================
  dart:async                                                _AsyncAwaitCompleter.completeError
  package:firebase_ml_vision/src/face_detector.dart         FaceDetector.processImage
  ===== asynchronous gap ===========================
  dart:async                                                _asyncThenWrapperHelper
  package:firebase_ml_vision/src/face_detector.dart         FaceDetector.processImage
  firebase_ml_vision_test.dart 608:49                       main.<fn>.<fn>.<fn>
```

**Comment by p30arena** [comment]
I observed that this doubles the output size of APK.
Was it the reason that you haven't had implemented contours in this plugin?

**Comment by bparrishMines** [comment]
Hi @p30arena 

We haven't added this feature yet because we are currently focusing on integration testing. Once we implement those, it will be easier for us to add additional features to the plugin. 

Also, to avoid automatically increasing the size of the APK, we would not include the line `api 'com.google.firebase:firebase-ml-vision-face-model:17.0.2'` in the `build.gradle` file. ML Kit models are quite large, so we allow the developer to choose what models to add. For example, if you want to use the `ImageLabeler` we instruct them to add the dependency themselves. [README](https://pub.dev/packages/firebase_ml_vision#-readme-tab-)

**Comment by p30arena** [comment]
@bparrishMines 
Hi there!
Isn't this going to be merged any time soon?

File: packages/firebase_ml_vision/README.md [hunk 1]
```diff
@@ -25,6 +25,19 @@ android {
 }
 ```
 
+### Android
```
Comment (bparrishMines) [inline packages/firebase_ml_vision/README.md]: This label is unnecessary since the last label was `### Android` above.

**Comment by bparrishMines** [comment]
> contours should never be null
> but I'm getting this error (maybe on iOS side?)
> 
> ```
> 00:07 +18 -1: /tmp/cirrus-ci-build/packages/firebase_ml_vision/test/firebase_ml_vision_test.dart: FirebaseVision FaceDetector processImage [E]                                                         
>   NoSuchMethodError: The method '[]' was called on null.
>   Receiver: null
>   Tried calling: []("allPoints")
>   dart:core                                                 Object.noSuchMethod
>   package:firebase_ml_vision/src/face_detector.dart 167:53  new Face._.<fn>
>   dart:collection                                           new LinkedHashMap.fromIterables
>   package:firebase_ml_vision/src/face_detector.dart 164:19  new Face._
>   package:firebase_ml_vision/src/face_detector.dart 85:22   FaceDetector.processImage
>   ===== asynchronous gap ===========================
>   dart:async                                                _AsyncAwaitCompleter.completeError
>   package:firebase_ml_vision/src/face_detector.dart         FaceDetector.processImage
>   ===== asynchronous gap ===========================
>   dart:async                                                _asyncThenWrapperHelper
>   package:firebase_ml_vision/src/face_detector.dart         FaceDetector.processImage
>   firebase_ml_vision_test.dart 545:49                       main.<fn>.<fn>.<fn>
> ```

Are you still running into this error? Could it be from not including the dependency in the app. I'm not sure if you are using the example app or another.

File: packages/firebase_ml_vision/README.md (hunk 1 above)
Comment (p30arena) [reply packages/firebase_ml_vision/README.md]: I removed the label

**Comment by p30arena** [comment]
> Thanks for the contribution!
> 
> Sorry for the delay. Everything looks good. I added a single comment.
> 
> Can you handle the merge conflicts, update the `pubspec.yaml`, and update the `CHANGELOG`? If not, I can also do these for you.

I merged the conflicts,
but please update the pubspec.yaml and CHANGELOG

Thanks

**Comment by p30arena** [comment]
> > contours should never be null
> > but I'm getting this error (maybe on iOS side?)
> > ```
> > 00:07 +18 -1: /tmp/cirrus-ci-build/packages/firebase_ml_vision/test/firebase_ml_vision_test.dart: FirebaseVision FaceDetector processImage [E]                                                         
> >   NoSuchMethodError: The method '[]' was called on null.
> >   Receiver: null
> >   Tried calling: []("allPoints")
> >   dart:core                                                 Object.noSuchMethod
> >   package:firebase_ml_vision/src/face_detector.dart 167:53  new Face._.<fn>
> >   dart:collection                                           new LinkedHashMap.fromIterables
> >   package:firebase_ml_vision/src/face_detector.dart 164:19  new Face._
> >   package:firebase_ml_vision/src/face_detector.dart 85:22   FaceDetector.processImage
> >   ===== asynchronous gap ===========================
> >   dart:async                                                _AsyncAwaitCompleter.completeError
> >   package:firebase_ml_vision/src/face_detector.dart         FaceDetector.processImage
> >   ===== asynchronous gap ===========================
> >   dart:async                                                _asyncThenWrapperHelper
> >   package:firebase_ml_vision/src/face_detector.dart         FaceDetector.processImage
> >   firebase_ml_vision_test.dart 545:49                       main.<fn>.<fn>.<fn>
> > ```
> 
> Are you still running into this error? Could it be from not including the dependency in the app. I'm not sure if you are using the example app or another.

This error is resolved

**Pull request merged**